### PR TITLE
Move insets to Redwood-only module

### PR DESCRIPTION
**Pull request by veyndan**
This is in preparation for allowing `UiConfiguration` to be provided within a Redwood-only & Compose UI-specific app.

Refs https://github.com/cashapp/redwood/issues/1054

### CODE REVIEW

File: redwood-widget-composeui/src/commonMain/kotlin/app/cash/redwood/widget/composeui/insets.kt [hunk 1]
```diff
@@ -13,11 +13,11 @@
...
-package app.cash.redwood.treehouse.composeui
+package app.cash.redwood.widget.composeui
 
 import androidx.compose.runtime.Composable
 import app.cash.redwood.ui.Margin
 
 /** Return the device's insets in screen density-independent pixels. */
 @Composable
-internal expect fun safeAreaInsets(): Margin
+public expect fun safeAreaInsets(): Margin
```
Comment (JakeWharton) [inline redwood-widget-composeui/src/commonMain/kotlin/app/cash/redwood/widget/composeui/insets.kt]: Does this revert once `RedwoodContent` exists in this module?

File: redwood-widget-composeui/build.gradle [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    publishLibraryVariants('release')
+  }
+
+  KmpTargets.addAllTargets(project, true /* skipJs */)
+
+  sourceSets {
+    commonMain {
+      dependencies {
+        implementation libs.jetbrains.compose.foundation
+        implementation projects.redwoodRuntime
```
Comment (JakeWharton) [inline redwood-widget-composeui/build.gradle]:
```suggestion
        api projects.redwoodRuntime
```
For `Margin`

**Review by JakeWharton — APPROVED** [review]

File: redwood-widget-composeui/src/commonMain/kotlin/app/cash/redwood/widget/composeui/insets.kt (hunk 1 above)
Comment (JakeWharton) [reply redwood-widget-composeui/src/commonMain/kotlin/app/cash/redwood/widget/composeui/insets.kt]: After looking at the other PR, I think this should move into `redwood-composeui` and not into a new module.

Comment (veyndan) [reply redwood-widget-composeui/src/commonMain/kotlin/app/cash/redwood/widget/composeui/insets.kt]: > Does this revert once RedwoodContent exists in this module?

Nope, cause we need it for `TreehouseContent`. Though if we make `TreehouseContent` a wrapper around `RedwoodContent`, we could revert it.

Comment (JakeWharton) [reply redwood-widget-composeui/src/commonMain/kotlin/app/cash/redwood/widget/composeui/insets.kt]: I think long term that should be the goal, but it's hard to reason about it right now. We can leave it as-is while we figure it out.

File: redwood-widget-composeui/build.gradle (hunk 2 above)
Comment (veyndan) [reply redwood-widget-composeui/build.gradle]: I've moved the insets over to `redwood-composeui`. `redwood-composeui` has an `api` dependency on `redwood-compose`, which has an `api` dependency on `redwood-runtime`, so we no longer need to explicitly specify a dependency on `redwood-runtime`.

**Pull request merged**